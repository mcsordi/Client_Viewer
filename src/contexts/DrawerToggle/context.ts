import { createContext } from 'react';
import { TNavDrawer } from '../../shared/types/NavDrawer';


type TToggle = {
    show: Boolean
    toggleState: () => void
    drawerOptions: TNavDrawer[]
    handleDrawerOptions: (navDrawer: TNavDrawer[]) => void
}

export const toggle = createContext({} as TToggle)
