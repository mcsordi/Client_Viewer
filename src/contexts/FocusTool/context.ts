import { createContext, Dispatch, SetStateAction } from "react";
type TFocusTool = {
    focus: boolean
    setFocus: Dispatch<SetStateAction<boolean>>
}

export const focusTool = createContext({} as TFocusTool)
