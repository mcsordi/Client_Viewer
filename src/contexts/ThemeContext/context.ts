import { createContext } from "react";

interface IThemeProvider {
    light: boolean,
    changeTheme: () => void
}

export const themeContext = createContext({} as IThemeProvider)

