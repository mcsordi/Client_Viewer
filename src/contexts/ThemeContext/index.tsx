import React, { useCallback, useState } from 'react';
import { themeContext } from './context.ts';
import { Children } from '../../shared/types/Children.ts';

export const ThemeContext: React.FC<Children> = ({ children }) => {
  const [light, setLight] = useState<boolean>(true);
  const changeTheme = useCallback(() => {
    setLight((old) => !old);
  }, []);
  return (
    <themeContext.Provider value={{ light, changeTheme }}>
      {children}
    </themeContext.Provider>
  );
};
