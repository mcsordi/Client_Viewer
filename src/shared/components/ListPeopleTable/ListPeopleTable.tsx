import React, { useContext } from 'react';
import { themeContext } from '../../../contexts/ThemeContext/context';
import { ConditionalTableRow } from '../ConditionalTableRow/ConditionalTableRow';
import { RiLoaderLine } from 'react-icons/ri';
import { TDataPeopleList } from '../../../services/people/PeopleService';
import { LoadComponent } from '../LoadComponent/LoadComponent';

type TListPeopleTable = {
  data: TDataPeopleList;
  isLoading: boolean;
  error: string;
};

export const ListPeopleTable: React.FC<TListPeopleTable> = ({
  data,
  isLoading,
  error,
}) => {
  const { light } = useContext(themeContext);

  return (
    <table
      className={`${light ? `bg-white` : `bg-zinc-700 text-white`} w-full mt-5 rounded-lg`}
    >
      <thead>
        <tr
          className={`${!light && `border-zinc-800`} border border-t-0 border-l-0 border-r-0`}
        >
          <th className="p-3  text-start">Ações</th>
          <th className="p-3  text-start">Nome</th>
          <th className="p-3  text-start">Email</th>
        </tr>
      </thead>
      {isLoading ? (
        <ConditionalTableRow>
          <LoadComponent />
        </ConditionalTableRow>
      ) : (
        <tbody>
          {data?.data?.length == 0 && (
            <ConditionalTableRow>
              Nenhum resultado foi encontrado na listagem
            </ConditionalTableRow>
          )}
          {error ? (
            <ConditionalTableRow>{error}</ConditionalTableRow>
          ) : (
            data?.data.map((person) => (
              <tr
                className={`${!light && `border-zinc-800`} border border-t-0 border-l-0 border-r-0 last:border-b-0`}
                key={person.id}
              >
                <td className="p-3 ">Ações</td>
                <td className="p-3 ">{person.fullName}</td>
                <td className="p-3 ">{person.email}</td>
              </tr>
            ))
          )}
        </tbody>
      )}
    </table>
  );
};
