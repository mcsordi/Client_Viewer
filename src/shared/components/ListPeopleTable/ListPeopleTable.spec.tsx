import { render, screen } from '@testing-library/react';
import { ListPeopleTable } from './ListPeopleTable';
import { themeContext } from '../../../contexts/ThemeContext/context';
import { toHaveClass } from '@testing-library/jest-dom/matchers';

const people = {
  data: [
    {
      fullName: 'Tiago Lancelotti',
      email: 'tiago@gmail.com',
      id: 1,
      cityId: 1,
    },
  ],
};

describe('<ListPeopleTable/>', () => {
  it('should render a table component with data and darkMode', () => {
    const { container } = render(
      <ListPeopleTable data={people} isLoading={false} error={undefined} />,
    );
    expect(container.firstChild).toHaveClass('bg-zinc-700 text-white');
    expect(container.firstChild.lastChild.hasChildNodes()).toBe(true);
    const personName = screen.getByText('Tiago Lancelotti');
    expect(personName).toBeInTheDocument();
  });
  it('should render a table component with data and light mode', () => {
    const changeTheme = () => {};
    const { container } = render(
      <themeContext.Provider value={{ light: true, changeTheme }}>
        <ListPeopleTable data={people} isLoading={false} error={undefined} />
      </themeContext.Provider>,
    );
    expect(container.firstChild).toHaveClass('bg-white');
    expect(container.firstChild.lastChild.hasChildNodes()).toBe(true);
  });
  it('should render a table component with no data', () => {
    render(
      <ListPeopleTable
        data={{ data: [] }}
        isLoading={false}
        error={undefined}
      />,
    );
    const noDataText = screen.getByText(
      'Nenhum resultado foi encontrado na listagem',
    );
    expect(noDataText).toBeInTheDocument();
  });
  it('should render a table component in loading', () => {
    const { container } = render(
      <ListPeopleTable data={people} isLoading={true} error={undefined} />,
    );
    expect(container.firstChild.lastChild.firstChild.firstChild).toHaveClass(
      'animate-spin text-2xl',
    );
  });
  it('should render a table component in error', () => {
    render(
      <ListPeopleTable data={people} isLoading={false} error={'hello there'} />,
    );
    const errorText = screen.getByText('hello there');
    expect(errorText).toBeInTheDocument();
  });
  it('should match a snapshot', () => {
    const { container } = render(
      <ListPeopleTable data={people} isLoading={false} error={''} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
