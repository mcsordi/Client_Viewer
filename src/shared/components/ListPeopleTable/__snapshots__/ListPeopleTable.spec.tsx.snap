// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListPeopleTable/> should match a snapshot 1`] = `
<table
  class="bg-zinc-700 text-white w-full mt-5 rounded-lg"
>
  <thead>
    <tr
      class="border-zinc-800 border border-t-0 border-l-0 border-r-0"
    >
      <th
        class="p-3  text-start"
      >
        Ações
      </th>
      <th
        class="p-3  text-start"
      >
        Nome
      </th>
      <th
        class="p-3  text-start"
      >
        Email
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      class="border-zinc-800 border border-t-0 border-l-0 border-r-0 last:border-b-0"
    >
      <td
        class="p-3 "
      >
        Ações
      </td>
      <td
        class="p-3 "
      >
        Tiago Lancelotti
      </td>
      <td
        class="p-3 "
      >
        tiago@gmail.com
      </td>
    </tr>
  </tbody>
</table>
`;
